util/codec/codec.go

```diff
@@ -352,6 +358,215 @@ func encodeHashChunkRowIdx(sc *stmtctx.StatementContext, row chunk.Row, tp *type
 	return
 }
 
+// HashChunkColumns writes the encoded value of each row's column, which of index `colIdx`, to h.
+func HashChunkColumns(sc *stmtctx.StatementContext, h []hash.Hash64, chk *chunk.Chunk, tp *types.FieldType, colIdx int, buf []byte, isNull []bool) (err error) {
+	var b []byte
+	column := chk.Column(colIdx)
+	rows := chk.NumRows()
+	switch tp.Tp {
+	case mysql.TypeTiny, mysql.TypeShort, mysql.TypeInt24, mysql.TypeLong, mysql.TypeLonglong, mysql.TypeYear:
+		i64s := column.Int64s()
```

Comment:
how about:
```go
for i := 0; i < rows; i++ {
    if column.IsNull(i) {
        h[i].Write(NilFlag)
        continue
    }
    h[i].Write(column.GetRaw(i))
}
```

Reply:
Do you mean no need to write the flag byte?

Reply:
Ignore the flag byte will generate a different hash value from that generated by `HashChunkRow`

Reply:
Okay. Maybe we can optimize this in another PR:
1. optimize the functions we used to calculate the hash value.
2. vectorize the way to calculate hash values for the outer table when performs a hash join.